tango/common/util.py

```diff
@@ -264,3 +266,37 @@ def utc_now_datetime() -> datetime:
 
 def local_timezone() -> Optional[tzinfo]:
     return datetime.now().astimezone().tzinfo
+
+
+def replace_steps_with_unique_id(o: Any):
+    from tango.step import Step
+
+    if isinstance(o, Step):
+        return {"type": "ref", "ref": o.unique_id}
+    if isinstance(o, (list, tuple, set)):
+        return o.__class__(replace_steps_with_unique_id(i) for i in o)
+    elif isinstance(o, dict):
+        return {key: replace_steps_with_unique_id(value) for key, value in o.items()}
+    else:
+        return o
+
+
+def jsonify(o: Any) -> Any:
+    """
+    Transform an arbitrary object into a JSON-serializable equivalent (if there is one)
+    in a deterministic way.
+    """
```

Comment:
I suspect you copied this from somewhere, so that's fine. But with a doc string like that I would also expect that it transforms single-item float tensors into `float`, because that's always an annoying problem too.